### PR TITLE
Add temporary API docs

### PR DESCRIPTION
**Pull request by thunderbiscuit**
## Description
This PR adds a directory that contains the required files to build the API docs for bdk-android and bdk-jvm. The docs need to be updated and published once a release.

## Notes to the reviewers
After getting comments on how specific-language API docs are useful in dev calls, I add these (albeit in a temporary fashion). We don't know how long it will be before the ability to use inline documentation is added to uniffi-rs, and so for the foreseeable future this is a good solution that will allow us to publish Kotlin API docs that work well and look good, and keep track of them in the same repository as the libraries themselves. I fully expect to remove this directory once the new workflows are in.

#### All Submissions:
* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)

### CODE REVIEW

**Comment by thunderbiscuit** [comment]
I also found a way to add samples (they show up as a new tab under the given object documentation page.

sample

**Comment by thunderbiscuit** [comment]
The current PR builds into [this website](https://thunderbiscuit.github.io/bdk-kotlin/).

File: api-docs/src/main/kotlin/org/bitcoindevkit/bdk.kt [hunk 1]
```diff
@@ -0,0 +1,453 @@
+package org.bitcoindevkit
+
+/**
+ * The cryptocurrency to act on.
+ */
+enum class Network {
+    /** Bitcoin's mainnet */
```
Comment (notmandatory) [inline api-docs/src/main/kotlin/org/bitcoindevkit/bdk.kt:7]: Just a consistency nit, but I think all javadocs lines should end with a period.

File: api-docs/src/main/kotlin/org/bitcoindevkit/bdk.kt [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+/**
+ * The address index selection strategy to use to derive an address from the wallet’s external descriptor.
+ *
+ * If you’re unsure which one to use, use `AddressIndex.NEW`.
+ */
+enum class AddressIndex {
+    /** Return a new address after incrementing the current descriptor index. */
+    NEW,
+
+    /** Return the address for the current descriptor index if it has not been used in a received transaction. Otherwise return a new address as with `AddressIndex.NEW`. Use with caution, if the wallet has not yet detected an address has been used it could return an already used address. This function is primarily meant for situations where the caller is untrusted; for example when deriving donation addresses on-demand for a public web page. */
```
Comment (notmandatory) [inline api-docs/src/main/kotlin/org/bitcoindevkit/bdk.kt:40]: Another nit, but I'd wrap this into multiple lines.

**Comment by notmandatory** [comment]
These docs look great, but for Java and Kotlin the docs can be a bit more expressive than in Rust. I recommend using this plugin: https://plugins.jetbrains.com/plugin/10389-kdoc-generator

It allows you do type `/** ` and return to get a basic template to fill out for the docs. The main difference from what you have now is it lets you also document the input/output parameters.  Example below. 

If you want to start with the short docs and fill in more details later I think what you have is fine for now.

```kotlin
/**
 * A blockchain client.
 *
 * @constructor Create a new blockchain client.
 *
 * @param the config for this blockchain client.
 */
class Blockchain(
    config: BlockchainConfig
) {
    /**
     * Broadcast a transaction.
     *
     * @param psbt the PSBT to broadcast.
     * @return the corresponding transaction id.
     */
    fun broadcast(psbt: PartiallySignedBitcoinTransaction): String {}

    /**
     * Get the current height of the blockchain.
     *
     * @return the block height.
     */
    fun getHeight(): UInt {}

    /**
     * Get the block hash of a given block.
     *
     * @param height the height of the block.
     * @return block hash base64 encoded.
     */
    fun getBlockHash(height: UInt): String {}
}

```

File: api-docs/src/main/kotlin/org/bitcoindevkit/bdk.kt (hunk 1 above)
Comment (thunderbiscuit) [reply api-docs/src/main/kotlin/org/bitcoindevkit/bdk.kt:7]: Ok cool I didn't know! I'll make sure they all do.

File: api-docs/src/main/kotlin/org/bitcoindevkit/bdk.kt (hunk 2 above)
Comment (thunderbiscuit) [reply api-docs/src/main/kotlin/org/bitcoindevkit/bdk.kt]: I brought all the long lines back, but didn't apply any particular rule. Do you have one in mind that you know is standard (length of line I mean?). I'm happy to apply a bit more consistently if there is a commonly agreed way to do KDocs/Javadoc documentation.

**Comment by thunderbiscuit** [comment]
Thanks for the plugin recommendation! Made my life much easier. All comments have been addressed, I think this is now ready for merging. We should consider filling them up a little bit before the next release with more samples, but otherwise the core is good IMO.

**Review by notmandatory — APPROVED** [review]
ACK 3f35a18d41fe0a2f3e7fe9345aa9cb9908d3f0ba

**Comment by notmandatory** [comment]
Thanks for doing all this! should help new folks get started, and also gives a great framework for filling in more docs.

**Pull request merged**